### PR TITLE
Add Sources view

### PR DESCRIPTION
**Pull request by jeff-phillips-18**

### CODE REVIEW

File: client/src/components/sources/sourcesToolbar.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+      type: dispatchTypes.SET_FILTER_VALUE,
+      filterValue
+    });
+  }
+
+  onValueKeyPress(keyEvent) {
+    const { filterValue, filterType } = this.props;
+
+    if (keyEvent.key === 'Enter' && filterValue && filterValue.length > 0) {
+      this.filterAdded(filterType, filterValue);
```
Comment (cdcabrera) [inline client/src/components/sources/sourcesToolbar.js]: Optional: ```truthy``` check on the ```filterValue.length```

File: client/src/components/sources/sourcesToolbar.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+      return (
+        <Filter.ValueSelector
+          filterValues={filterType.filterValues}
+          currentValue={filterValue}
+          placeholder={filterType.placeholder}
+          onFilterValueSelected={this.filterValueSelected}
+        />
+      );
+    } else {
+      return (
```
Comment (cdcabrera) [inline client/src/components/sources/sourcesToolbar.js]: Optional: Same comment about the ```else```s with ```return```s, may be able to get away with out them and the encompassing curly braces

File: client/src/components/sources/sourcesToolbar.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+          />
+          <Sort.DirectionSelector
+            isNumeric={sortType.isNumeric}
+            isAscending={sortAscending}
+            onClick={() => this.toggleCurrentSortDirection()}
+          />
+        </Sort>
+      );
+    } else {
+      return null;
```
Comment (cdcabrera) [inline client/src/components/sources/sourcesToolbar.js]: Optional: Same comment about ```else```s and ```return```s

File: client/src/components/sources/sourcesToolbar.js [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    );
+  }
+
+  renderCounts() {
+    const { activeFilters, totalCount, filteredCount } = this.props;
+
+    return (
+      <h5>
+        {activeFilters && activeFilters.length > 0
+          ? filteredCount + ' of '
```
Comment (cdcabrera) [inline client/src/components/sources/sourcesToolbar.js]: Optional: ```truthy``` check
Optional: string literal on 
```js
`${filterCount} of`
```

File: client/src/components/sources/sourcesToolbar.js [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+          : null}
+        {totalCount + ' Result'}
```
Comment (cdcabrera) [inline client/src/components/sources/sourcesToolbar.js]: Optional: string literal might help with the pluralization... or not, having typed it out it's a little verbose
```js
{
`${totalCount} ${totalCount>1 ? 'Result': 'Results'}`
}
```

File: client/src/services/sourcesApi.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
+    return fetch('http://localhost:4000/api/v1/sources/').then(response => {
+      if (response.ok) {
         return response.json();
-      })
-      .catch(error => {
-        return error;
-      });
+      } else {
+        throw new Error(response.statusText);
+      }
```
Comment (cdcabrera) [inline client/src/services/sourcesApi.js]: +1 good catch on removing the ```catch``` and allowing it to pass through to the action

File: client/src/components/sources/sources.js [hunk 7]
```diff
@@ -1,48 +1,166 @@
...
+    } else if (loadError) {
+      return (
+        <EmptyState>
+          <Alert type="danger">
+            <span>Error retrieving sources: {errorMessage}</span>
+          </Alert>
+        </EmptyState>
+      );
+    } else if (sources && sources.length > 0) {
+      let filteredSources = this.filterSources(sources);
```
[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply client/src/components/sources/sources.js]: Yeah, I miss `_.size`

**Pull request merged**